stdlib/src/utils/static_tuple.mojo

```diff
@@ -254,3 +314,142 @@ struct StaticTuple[element_type: AnyRegType, size: Int](Sized):
         var base_ptr = Pointer[Self.type].address_of(self.array).address
         var ptr = __mlir_op.`pop.array.gep`(base_ptr, Int(0).value)
         return Pointer(ptr)
+
+
+# ===----------------------------------------------------------------------===#
+# Array
+# ===----------------------------------------------------------------------===#
+
+
+@value
+struct Array[ElementType: CollectionElement, size: Int](Sized):
+    """A statically sized tuple type which contains elements of homogeneous types.
+
+    Parameters:
+        ElementType: The type of the elements in the tuple.
+        size: The size of the tuple.
+    """
+
+    alias type = __mlir_type[
+        `!pop.array<`, size.value, `, `, Self.ElementType, `>`
+    ]
+    var array: Self.type
+    """The underlying storage for the static tuple."""
+
+    @always_inline
+    fn __init__(inout self):
+        """Constructs an empty (undefined) tuple."""
+        _static_tuple_construction_checks[size]()
+        self.array = __mlir_op.`kgen.undef`[_type = Self.type]()
+
+    @always_inline
+    fn __init__(inout self, *elems: Self.ElementType):
+        """Constructs a static tuple given a set of arguments.
+
+        Args:
+            elems: The element types.
+        """
+        debug_assert(
+            0 < len(elems) < size,
+            "number of variadic elements must be within range",
+        )
+        _static_tuple_construction_checks[size]()
+        self = Self()
+        var i = 0
+        for e in elems:
+            self[i] = e[]
+            i += 1
+
+    @always_inline
+    fn __init__(inout self, values: VariadicList[Self.ElementType]):
+        """Creates a tuple constant using the specified values.
+
+        Args:
+            values: The list of values.
+        """
+        _static_tuple_construction_checks[size]()
+        debug_assert(
+            0 < len(values) < size, "size of `values` must be within range"
```

Comment:
There is also `debug_assert(size == len(lst), ...)` inside `_create_array_collection_elem`. Maybe this should just be a comment to say `len(values)` is checked there?